analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -496,7 +496,8 @@ private static bool IsDeclaredInPartialClass(ISymbol methodSymbol)
 
             private static bool IsRemovable(ISymbol symbol) =>
                 symbol is { IsImplicitlyDeclared: false, IsVirtual: false }
-                && !symbol.GetAttributes().Any()
+                && !symbol.IsSerializableMember()
+                && !symbol.GetAttributes().Any(x => !x.AttributeClass.Is(KnownType.System_NonSerializedAttribute))
```

Comment:
Isn't this redundant based on the internal implementation of `IsSerializable`?

Reply:
If I delete either it will turn the UTs red. One of the conditions is negated the other isn't (or more like double-negated).